src/main/java/org/opensearch/ad/model/AnomalyDetector.java

```diff
@@ -465,12 +516,15 @@ public static AnomalyDetector parse(
                 case USER_FIELD:
                     user = User.parse(parser);
                     break;
+                case ML_MODEL_ID_FIELD:
+                    mlModelId = parser.text();
+                    break;
                 default:
                     parser.skipChildren();
                     break;
             }
         }
-        return new AnomalyDetector(
+        AnomalyDetector anomalyDetector = new AnomalyDetector(
```

Comment:
What is the need for this? Why can't we use the new constructor?

Reply:
If you check below, there is a line `anomalyDetector.setMlModelId(mlModelId);`. Because of the issue mentioned in this PR's "Description - Future tasks and improvements #5" and to support BWC, I think this is the best way to put it now.